### ISSUE TITLE
ESLint not respecting ignore rules correctly

### ISSUE THREAD

**Issue by moltar** [eslint/eslint#12087]
**Tell us about your environment**

* **ESLint Version:** v6.1.0
* **Node Version:** v10.15.3
* **npm Version:** 6.10.3

**What parser (default, Babel-ESLint, etc.) are you using?**

See [`.eslintrc.js`](https://github.com/moltar/tsnm/blob/feature/eslint/.eslintrc.js) for details.

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint'],
  extends: [
    'eslint:recommended',
    'plugin:@typescript-eslint/eslint-recommended',
    'airbnb-base',
    'plugin:@typescript-eslint/recommended',
    'prettier',
    'prettier/@typescript-eslint'
  ],
  rules: {
    'import/prefer-default-export': 'off',
    '@typescript-eslint/explicit-function-return-type': 'off',
  },
  settings: {
    'import/resolver': {
      typescript: {
        directory: './tsconfig.node.json'
      }
    }
  },
  overrides: [
    {
      env: {
        browser: true,
      },
      files: ['src/node_modules/browser-*/**/*.ts'],
    },
    {
      env: {
        node: true,
      },
      files: ['src/node_modules/node-*/**/*.ts'],
    },
  ]
}

```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

See this issue for reproducible example: https://github.com/moltar/tsnm/issues/2

```bash
eslint --ignore-path .gitignore --ext .js,.ts src/
```

**What did you expect to happen?**

Rules to be applied and linter to run.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint ignores all of the files and directories.

**Are you willing to submit a pull request to fix this bug?**

Yes.

**Comment by platinumazure** [comment]
Sorry for the auto-close. We've had a lot of issues and a lot of busy maintainers.

Looking at the linked issue, I think I agree that the .eslintignore definitely does not follow .gitignore anymore and we should change the docs.

I think in your case, you could unignore `!src/node_modules/**` and that will hopefully get what you need. (Yes, apologies, you might need an .eslintignore file for now.)

We should definitely have an issue opened here about .eslintignore not following .gitignore, so the team can decide whether we want to try to bring that capability back, or change the docs.
